src/components/AdminPageTable/AdminPageTable.jsx

```diff
@@ -1,43 +1,59 @@
 /* eslint-disable react/jsx-key */
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { useTable, usePagination } from 'react-table';
-import { Table, Tr, Td, Tbody } from '@chakra-ui/react';
+import { useTable, usePagination, useSortBy } from 'react-table';
+import { Table, Tr, Td, Tbody, Spinner } from '@chakra-ui/react';
 import AdminPageHeader from './AdminPageHeader';
 import Pagination from './Pagination';
 import CellStructure from './AdminPageStructure';
 
 /* eslint-enable react/destructuring-assignment, react/prop-types */
 
-const AdminPageTable = ({ tableData, checked, setChecked, allChecked, setAllChecked }) => {
+const AdminPageTable = ({
+  tableData,
+  pageCount: controlledPageCount,
+  checked,
+  isLoading,
+  setChecked,
+  allChecked,
+  setAllChecked,
+  setFetchSettings,
+}) => {
   const columns = useMemo(
     () => CellStructure(checked, setChecked, allChecked, setAllChecked),
     [checked, setChecked, allChecked, setAllChecked],
   );
-  const data = useMemo(() => tableData);
-
   const {
     getTableProps,
     getTableBodyProps,
     headerGroups,
     rows,
     page,
     prepareRow,
+    pageCount,
```

Comment:
⚠️ **[eslint]** <[no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'pageCount' is assigned a value but never used.